assignment3/Assignment3.java

```diff
@@ -0,0 +1,53 @@
+package assignment3;
+
+import org.junit.Test;
+import static org.junit.Assert.assertEquals;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class Assignment3 {
+
+    @Test
+    public void testEmptyDict() {
+        char[][] grid = new char[][] {
+            {'a', 'a', 'r'},
+            {'t', 'c', 'd'}
+        };
+        WordDictionary dict = new WordDictionary();
+        Set<String> expected = new HashSet<>();
+        Set<String> words = new GridWordSearcher(grid.length, grid[0].length, grid, dict).findWords();
+        assertEquals("Set of words is not empty", words, expected);
+    }
+
+    @Test
+    public void testGivenExample() {
+        char[][] grid = new char[][] {
+                {'a', 'a', 'r'},
+                {'t', 'c', 'd'}
+        };
+        WordDictionary dict = new WordDictionary("car", "card", "cat", "cart");
+        Set<String> expected = new HashSet<String>() {{
+            add("car");
+            add("card");
+            add("cat");
+        }};

```

Comment:
This is fine, but if I was writing this at Google I would use Guava (another open source library https://github.com/google/guava) and write:
ImmutableSet.of("car", "card", "cat").

Or I would use Truth and write:
assertThat(words).containsExactly("car", "card", "cat")